test_vm/src/lib.rs

```diff
@@ -75,6 +75,39 @@ use std::ops::Add;
 
 pub mod util;
 
+/// An abstract VM that is injected into integration tests
+pub trait VM<BS: Blockstore> {
+    /// Returns the underlying blockstore of the VM
+    fn blockstore(&self) -> Box<&BS>;
+
+    /// Get the state root of the specified actor
+    fn state_root(&self, address: &Address) -> Option<Cid>;
+
+    /// Get the current chain epoch
+    fn epoch(&self) -> ChainEpoch;
+
+    /// Get the balance of the specified actor
+    fn balance(&self, address: &Address) -> TokenAmount;
+
+    /// Get the ID for the specified address
+    fn resolve_id_address(&self, address: &Address) -> Option<Address>;
+
+    /// Send a message between the two specified actors
+    /// TODO: add a generic parameter to MessageResult to allow an extra value to be returned
+    fn execute_message(
+        &self,
+        from: &Address,
+        to: &Address,
+        value: &TokenAmount,
+        method: MethodNum,
+        params: Option<IpldBlock>,
+    ) -> Result<MessageResult, TestVMError>;
+
+    /// Sets the epoch to the specified value
+    fn set_epoch(&self, epoch: ChainEpoch);
```

Comment:
Mock behaviour such as this may be hard to implement on different VM implementations.